### PR TITLE
docs: How CozyClient store works

### PR DESCRIPTION
**Pull request by Crash--**
Try to explain how queries and documents slice are updated

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+action for a simple `get` but `receiveMutationResult` when we mutate 
+something. 
+
+Focus on `receiveMutationResult`: 
+
+Our two slices `documents` and `queries` listen actions and do some specific work on
+ `isReceivingMutationResult` action.
+
+`documents`: If the `_id` of the mutated document is not present, then we add the document. 
+If the `_id` is already there, then we update the content of the document we the fresh data
```
Comment (JF-Cozy) [inline docs/architecture.md:155]:
```diff
- of the document we the fresh data
+ of the document with the fresh data
```

File: docs/architecture.md [hunk 2]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+- it takes the mutated documents (newly created or updated) 
+- it converts the initial `query` to a "`js predicate`" (thanks to sift javascript librairy)
+- For each queries already in the `slice` it runs this `js predicate` and detect if the query
+is concerned by the mutation
+- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
+document
+(for now the work is done mainly in queries.js/updateData())
+
+So in our previous exemple our `todoslist` is concerned by the addition of the new todo, then 
+the `id` is add to `todolist` data, then the component linked will be refreshed with this new
```
Comment (JF-Cozy) [inline docs/architecture.md:171]:
```diff
- the `id` is add to `todolist` data
+ the `id` is added to `todolist` data
```

**Review by JF-Cozy — APPROVED** [review]

File: docs/architecture.md [hunk 3]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+If the `_id` is already there, then we update the content of the document we the fresh data
+(for now the work is done in extractAndMergeDocument() method). 
+
+So if your app is linked to the documents store via getDocumentFromStore() for instance
+your app will have the updated value. 
+
+`queries` has an autoUpdater mechanism that does something we can explain this way:
+- it takes the mutated documents (newly created or updated) 
+- it converts the initial `query` to a "`js predicate`" (thanks to sift javascript librairy)
+- For each queries already in the `slice` it runs this `js predicate` and detect if the query
```
Comment (trollepierre) [inline docs/architecture.md:164]:
```suggestion
- For each query already in the `slice` it runs this `js predicate` and detects if the query
```

File: docs/architecture.md [hunk 4]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+is concerned by the mutation
+- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
+document
+(for now the work is done mainly in queries.js/updateData())
+
+So in our previous exemple our `todoslist` is concerned by the addition of the new todo, then
```
Comment (trollepierre) [inline docs/architecture.md:170]:
```suggestion
So in our previous example our `todoslist` is concerned by the addition of the new todo, then 
```

**Review by trollepierre — APPROVED** [review]

File: docs/architecture.md [hunk 5]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+After the call to `save` your `TodoListComponent` will be re-rendered 
+with the `New TODO`. 
+
+How does it works?
+
+When a `query` is resolved, CozyClient dispatches a `receiveQueryResult`
+action for a simple `get` but `receiveMutationResult` when we mutate 
+something. 
+
+Focus on `receiveMutationResult`:
```
Comment (paultranvan) [inline docs/architecture.md:149]:
```suggestion
### Focus on `receiveMutationResult`: 
```

File: docs/architecture.md [hunk 6]
```diff
@@ -118,3 +117,60 @@ The glue between the Redux store and the UI is done via ObservableQuery.
...
+`documents`: If the `_id` of the mutated document is not present, then we add the document. 
+If the `_id` is already there, then we update the content of the document we the fresh data
+(for now the work is done in extractAndMergeDocument() method). 
+
+So if your app is linked to the documents store via getDocumentFromStore() for instance
+your app will have the updated value. 
+
+`queries` has an autoUpdater mechanism that does something we can explain this way:
+- it takes the mutated documents (newly created or updated) 
+- it converts the initial `query` to a "`js predicate`" (thanks to sift javascript librairy)
```
Comment (paultranvan) [inline docs/architecture.md:163]:
```suggestion
- it converts the initial `query` to a "`js predicate`" (thanks to the [sift library](https://github.com/crcn/sift.js/))
```

**Comment by paultranvan** [comment]
Thank you for the additional explanations :clap:

**Comment by paultranvan** [comment]
@Crash-- will you have time to finish this PR? Or do you need some help?

**Comment by Crash--** [comment]
done, thanks for the review :)

**Pull request merged**